pkg/ingester/transfer.go

```diff
@@ -266,19 +287,30 @@ func (i *Ingester) findTargetIngester(ctx context.Context) (*ring.IngesterDesc,
 		return &ingesters[0], nil
 	}
 
-	deadline := time.Now().Add(i.cfg.SearchPendingFor)
+	deadline := time.NewTimer(i.cfg.SearchPendingFor)
+	defer deadline.Stop()
+
+	ticker := time.NewTicker(i.cfg.SearchPendingFor / pendingSearchIterations)
+	defer ticker.Stop()
+
 	for {
-		ingester, err := findIngester()
-		if err != nil {
-			level.Debug(util.Logger).Log("msg", "Error looking for pending ingester", "err", err)
-			if time.Now().Before(deadline) {
-				time.Sleep(i.cfg.SearchPendingFor / pendingSearchIterations)
+		select {
+		case <-ticker.C:
```

Comment:
If we have an outer loop doing retries, do we still want this inner loop?

Reply:
I've also deprecated one of the flags that was used here (`ingester.search-pending-for`), which is replaced by `ingester.max-transfer-retries`.  I made the max backoff 5s, so with 10 retries this should approximate the default 10s search time from the previous flag.